src/server/courseInstance/courseInstance.service.ts

```diff
@@ -171,4 +185,53 @@ export class CourseInstanceService {
       .addOrderBy('"courseNumber"', 'ASC')
       .getMany() as Promise<ScheduleViewResponseDTO[]>;
   }
+
+  /**
+   * Accepts a list of instructorIds and a courseInstanceId, assigns the
+   * faculty in the order they appear in the array, then returns the instructor
+   * data formatted for display
+   */
+  public async assignInstructors(
+    courseInstanceId: string,
+    instructors: InstructorRequestDTO[]
+  ): Promise<InstructorResponseDTO[]> {
+    // Pull the instructors from the database to ensure that they actually
+    // exist. Using .map() instead of .findByIds() to ensure order is unchanged
+    const dbInstructors = await Promise.all(
+      instructors.map(
+        ({ id }) => this.facultyRepository.findOneOrFail(id)
+      )
+    );
+    const courseInstance = await this.courseInstanceRepository
+      .findOneOrFail(courseInstanceId);
+    const updates = dbInstructors.map(
+      (faculty, order): FacultyCourseInstance => (
+        this.instructorRepository.create({
+          courseInstance,
+          faculty,
+          order,
+        }))
+    );
+
+    // Adding the list of updates to the course isntances, instead of saving
+    // them directly, to ensure that the new list of instructors replaces the
+    // old one.
+    const instructorUpdate = await this.courseInstanceRepository.save(
+      {
+        ...courseInstance,
+        facultyCourseInstances: updates,
+      }
+    );
+    if (instructorUpdate) {
+      return instructorUpdate
+        .facultyCourseInstances
+        .map(({ faculty, order }) => ({
+          id: faculty.id,
+          notes: faculty.notes,
+          displayName: `${faculty.lastName}, ${faculty.firstName}`,
+          order,
+        }));
+    }
+    throw new Error('Unable to update instructor list');
```

Comment:
Looking at this again, I don't think this `throw` is really necessary at all, since the only case where  `instructorUpdate` wouldn't exist is if `save()` threw an error, in which case we wouldn't hit this line anyway. Removed in bfd26a7d.

Reply:
That makes sense.